runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -95,7 +95,7 @@ InterpreterEmulator::maintainStackForGetField()
             {
             TR::VMAccessCriticalSection dereferenceKnownObjectField(comp()->fej9());
```

Comment:
This piece is also problematic. It is executed with VMaccess in hand to prevent GC from moving objects. However, the server will send a few messages to the client and the client does not hold VMaccess throughout this sequence of operations. Somehow we need to create a big function out of this code and execute it atomically (that is under VMaccess) at the client.

Reply:
I think it can be done as a message that sends to the client the mirror (for `_calleeMethod->getDeclaringClassFromFieldOrStatic()`) the `baseObjectIndex`, the `cpIndex`, the `fieldOffset` and receives a KOT index for the `fieldAddress`. However, if a new entry was added the KOT at the client, the server must do the same.